test/src/kbasesearchengine/test/main/NarrativeInfoDecoratorTest.java

```diff
@@ -372,6 +377,66 @@ public void listTypes() throws Exception {
                 "type", dummyOutput)));
     }
 
+    @Test
+    public void searchObjectsWithDeletedWs() throws Exception {
+        final NarrativeInfoDecorator nid = setUpSearchObjectsWithDeletedWs();
+        final Pagination pag = new Pagination().withCount(3L).withStart(0L);
+        final SearchObjectsInput dummyInput = new SearchObjectsInput()
+                .withPostProcessing(new PostProcessing().withAddNarrativeInfo(1L))
+                .withPagination(pag);
+
+        final ArrayList<ObjectData> expectedObjs = new ArrayList<>();
+        expectedObjs.add(new ObjectData().withGuid("WS:4/1/7").withCreator("user"));
+        expectedObjs.add(new ObjectData().withGuid("WS:4/2/1").withCreator("user"));
+        expectedObjs.add(new ObjectData().withGuid("WS:4/1/7").withCreator("user"));
```

Comment:
It would be good to have a test where the requested count is not fulfilled (because of deleted workspaces, but simulated of course) and subsequent calls to reSearchObjects() and combineWithOtherSearchObjectOutput() are made in the searchObjects() while loop.